### PR TITLE
Issue #14 & #26 and fixes

### LINKED ISSUE #14: Docs: explain health vs refusing to sign new states
Explain why any new valid state transition would be signed, and how the balances are distributed is tracked in the `health`

In short, the idea is that how the balances would be distributed can be subjective - different validators may receive different events if there is something wrong with the network connections

While signing an invalid state transition is clearly a violation - it clearly means malicious behavior (trying to go back on what you paid); of course, you can create valid state transitions which pay out to yourself (or to others who haven't legtimately earned), but as the leader, the funds are yours anyway (would be much easier to just [cancel the channel](https://github.com/adexnetwork/adex-protocol#closing-a-campaign))

-----

from a technical standpoint, this makes implementation easier

health represents the comparison between the latest approved `NewState` and our own `stateTree`

while the state transition checks previous approved `NewState` vs candidate `NewState`

------

**However,** dramatically poor health can and should result in a refusal to sign (see https://github.com/AdExNetwork/adex-validator/issues/47)

### PR DESCRIPTION
**Pull request by elpiel**
- [x] validator stack - Sentry API Resolves _Partially_  #26 
- [x] Refusal to sign - Resolves #14

### CODE REVIEW

**Review by samparsky — APPROVED** [review]

File: components/validator-stack.md [hunk 1]
```diff
@@ -26,11 +27,83 @@ The Sentry is a stateless microservice, which means it can scale horizontally. T
...
+
+##### Verified Payload
+
+The `Verified Payload` contains the decoded `Signer address` and `Payload` from the `AUTHENTICATION` header.
+
+#### Authentication token and Authorization
+
+Using the `Verified Payload` we perform 2 more checks:
+
+##### 1. Was this Authentication toke intended for this validator?
```
Comment (IvoPaunov) [inline components/validator-stack.md:98]: token

File: components/validator-stack.md [hunk 2]
```diff
@@ -26,11 +27,83 @@ The Sentry is a stateless microservice, which means it can scale horizontally. T
...
+##### Decoding the Payload
+
+We decode the `Payload` part of the `AUTHORIZATION` header with `base64` again using URL-safe character set without padding configuration.
 
-@TODO explain what the flow is FOR; all processes that require the validator stack
+Decoding the `Payload` should result in a JSON string with the following fields:
+
+* `id` - string - "0x" prefixed address of the validator this request is intended for
+* `era` - number - `current Date & Time milliseconds / 60 000` or simply the current date and time minutes passed from epoch in milliseconds
+* `address` - string - "0x" prefixed address (TODO: what is it used for?)
```
Comment (IvoPaunov) [inline components/validator-stack.md:83]: maybe remove the TODO?

File: components/validator-stack.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 
-## In progress
+#### Reference implementations of the validator stack
 
-#### This document is currently a work in progress.
-
-#### However, there is an actual reference implementation of the validator stack here: https://github.com/adexnetwork/adex-validator
+* *JavaScript (Node.js)*: https://github.com/adexnetwork/adex-validator
+* *Rust*: https://github.com/AdExNetwork/adex-validator-stack-rust
+    As of October 2020 the Rust implementation of Sentry is lacking behind the JavsScript implementation.
```
Comment (IvoPaunov) [inline components/validator-stack.md:7]: JavsScript typo

**Review by IvoPaunov — CHANGES_REQUESTED** [review]
see the comments - some typos

File: components/validator-stack.md (hunk 1 above)
Comment (elpiel) [reply components/validator-stack.md]: Fixed, tnx!

File: components/validator-stack.md (hunk 2 above)
Comment (elpiel) [reply components/validator-stack.md]: Fixed, tnx!

File: components/validator-stack.md (hunk 3 above)
Comment (elpiel) [reply components/validator-stack.md]: Fixed, tnx!

**Review by IvoPaunov — APPROVED** [review]
see the comments - some typos

**Pull request merged**